### PR TITLE
Fix WebView SDK28 issue

### PR DESCRIPTION
**Pull request by rpdome**
See: https://developer.android.com/about/versions/pie/android-9.0-changes-28#web-data-dirs

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java [hunk 1]
```diff
@@ -178,11 +180,32 @@ private void extractState(final Bundle state) {
...
     @Override
     protected void onCreate(final Bundle savedInstanceState) {
         final String methodName = "#onCreate";
         super.onCreate(savedInstanceState);
 
+        // Must be invoked before WebvView in common_activity_authentication is inflated.
+        // See https://developer.android.com/about/versions/pie/android-9.0-changes-28#web-data-dirs for more info.
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            try {
+                WebView.setDataDirectorySuffix(getProcessName(getApplicationContext()));
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java]: should we check if process name is empty and only set if it's `auth` , i.e. for MSAL local auth, what would be the process name?

Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java]: fixed.

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/WebViewUtil.java [hunk 2]
```diff
@@ -23,26 +23,75 @@
...
+     * */
+    @SuppressLint("NewApi")
+    public static void setDataDirectorySuffix(@NonNull final Context context){
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            try {
+                if (ProcessUtil.isRunningOnAuthService(context)) {
+                    WebView.setDataDirectorySuffix(AuthServiceProcess);
+                }
+            } catch (final IllegalStateException e) {
+                // If webView is already initialized, this will be thrown.
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/ui/webview/WebViewUtil.java]: may be add a `warn` log here

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/WebViewUtil.java [hunk 3]
```diff
@@ -23,26 +23,75 @@
...
             final CookieSyncManager syncManager = CookieSyncManager.createInstance(context);
-            syncManager.startSync();
             cookieManager.removeAllCookie();
-            syncManager.stopSync();
             syncManager.sync();
         }
     }
+
+    /**
+     * Clear session cookies from embedded webview.
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/ui/webview/WebViewUtil.java]: We should probably javadoc that this is blocking call and so should not be called on UI thread

**Pull request merged**